--- origsrc/Premake-3.5/Src/gnu_cpp.c	2007-10-22 11:08:35.000000000 -0500
+++ src/Premake-3.5/Src/gnu_cpp.c	2007-12-01 21:35:19.937500000 -0600
@@ -84,7 +84,7 @@
 
 		/* Write C flags */
 		io_print("  CFLAGS += $(CPPFLAGS) $(TARGET_ARCH)");
-		if (prj_is_kind("dll") && !os_is("windows"))
+		if (prj_is_kind("dll") && !os_is("windows") && !os_is("cygwin"))
 			io_print(" -fPIC");
 		if (!prj_has_flag("no-symbols"))
 			io_print(" -g");
--- origsrc/Premake-3.5/Src/os.c	2006-06-23 09:41:08.000000000 -0500
+++ src/Premake-3.5/Src/os.c	2007-12-01 21:35:19.953125000 -0600
@@ -33,6 +33,8 @@
 	os = "bsd";
 #elif defined(__APPLE__) && defined(__MACH__)
 	os = "macosx";
+#elif defined(__CYGWIN__)
+	os = "cygwin";
 #else
 	os = "windows";
 #endif
--- origsrc/Premake-3.5/Src/os.h	2007-03-05 19:41:59.000000000 -0600
+++ src/Premake-3.5/Src/os.h	2007-12-01 21:35:19.968750000 -0600
@@ -17,7 +17,7 @@
 
 /* Determine the current OS. I'm not sure how to reliably detect Windows
  * but since it is the most common I use is as the default */
-#if defined(__linux__)
+#if defined(__linux__) || defined(__CYGWIN__)
 #define PLATFORM_POSIX 1
 #elif defined(__FreeBSD__) || defined(__NetBSD__) || defined(__OpenBSD__)
 #define PLATFORM_POSIX 1
--- origsrc/Premake-3.5/Src/project.c	2007-11-08 20:01:45.000000000 -0600
+++ src/Premake-3.5/Src/project.c	2007-12-01 21:35:19.968750000 -0600
@@ -774,6 +774,29 @@
 		}
 	}
 
+	else if (os_is("cygwin"))
+	{
+		if (matches(cfg->kind, "lib"))
+		{
+			if (cfg->prefix == NULL)
+				strcat(buffer, "lib");
+			strcat(buffer, filename);
+			extension = "a";
+		}
+		else if (matches(cfg->kind, "dll"))
+		{
+			if (cfg->prefix == NULL)
+				strcat(buffer, "cyg");
+			strcat(buffer, filename);
+			extension = "dll";
+		}
+		else
+		{
+			strcat(buffer, filename);
+			extension = "exe";
+		}
+	}
+
 	else if (os_is("macosx") && matches(cfg->kind, "dll"))
 	{
 		if (cfg->prefix == NULL)
